classes/external/get_data_merge_requests.php

```diff
@@ -0,0 +1,150 @@
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class get_data_merge_requests extends \external_api {
+    /**
+     * Webservice tool_mergeusers_get_data_merge_request parameters
+     *
+     * @return external_function_parameters
+     */
+    public static function execute_parameters(): external_function_parameters {
+        return new external_function_parameters([
+            'mergeusers' => new external_single_structure([
+                'removeuserfield' => new external_value(
+                    PARAM_TEXT, 'Remove user field', VALUE_OPTIONAL),
+                'removeuservalue' => new external_value(
+                    PARAM_RAW, 'Remove user value', VALUE_OPTIONAL),
+                'removeuserid' => new external_value(
+                    PARAM_INT, 'Identifier of the remove user id', VALUE_OPTIONAL),
+                'keepuserfield' => new external_value(
+                    PARAM_TEXT, 'Keep user field', VALUE_OPTIONAL),
+                'keepuservalue' => new external_value(
+                    PARAM_RAW, 'Keep user value', VALUE_OPTIONAL),
+                'keepuserid' => new external_value(
+                    PARAM_INT, 'Identifier of the keep user id', VALUE_OPTIONAL),
+                'id' => new external_value(
+                    PARAM_INT, 'Identifier of the merge request', VALUE_OPTIONAL),
+                'status' => new external_value(
+                    PARAM_INT, 'Status of the merge request', VALUE_OPTIONAL)
+            ])
+        ]);
+    }
+    /**
+     * Return data of the custom queue of the merge request
+     *
+     * @param $mergeusers id of the merge users request
+     *
+     * Return array with data of the custom queue of the merge request
+     * @param $id
+     * @param $removeuserfield Remove user field
+     * @param $removeuserid Remove user id
+     * @param $removeuservalue Remove user value
+     * @param $keepuserfield Keep user field
+     * @param $keepuserid Keep user id
+     * @param $keepuservalue Keep user value
+     * @param $timeadded Time creation
+     * @param $timemodified Time modified
+     * @param $status Status
+     * @param $retries Number of retries
+     * @param $log Log
+     * 
+     */
+    public static function execute(array $mergeusers) {
+        global $DB;
+        // Validate all of the parameters.
+        $params = array();
+        $params = self::validate_parameters(self::execute_parameters(),
+                                                array('id' => $mergeusers->id));
+        $tablemr = merge_request::TABLE_MERGE_REQUEST;
+        $sql = "SELECT
+                    id, removeuserfield, removeuservalue, removeuserid,
+                    keepuserfield, keepuservalue, keepuserid,
+                    timeadded, timemodified, status, retries, log
+                FROM
+                   {".merge_request::TABLE_MERGE_REQUEST."}
+                WHERE
```

Comment:
you do not need to add here `WHERE (1=1)`. Leave the base $sql without this clause. Then, if some filtering applies, add `WHERE ` as a prefix to the the where clause.

Reply:
the problem is that you don't know if you have one or more filters.

Reply:
yes, you will know.

instead of just appending elements to the `$sql`, you can just build an array of conditions, like `$whereclauses[] = 'removeuserfield = ?';`. And then, at the end, evaluate if there are where clauses, like:

```php
if (count($wherecluases) > 0) {
    $sql .= 'WHERE ' . implode(' AND ', $whereclauses);
}
```

And you will have a cleaner result.